skglm/penalties/separable.py

```diff
@@ -466,3 +467,45 @@ def is_penalized(self, n_features):
     def generalized_support(self, w):
         """Return a mask with non-zero coefficients."""
         return w != 0
+
+
+class SLOPE(BasePenalty):
+    """Sorted L-One Penalized Estimation (SLOPE) penalty.
+
+    References
+    ----------
+    .. [1] B. Malgorzata, E. van den Berg, C. Sabatti, W. Su, E. Candes
+    "SLOPE - Adaptive Variable Selection via Convex Optimization",
+    The Annals of Applied Statistics 9 (3): 1103–40
+    https://doi.org/10.1214/15-AOAS842
+    """
+
+    def __init__(self, alphas):
+        self.alphas = alphas
+
+    def get_spec(self):
+        spec = (
+            ('alphas', float64[:]),
+        )
+        return spec
+
+    def params_to_dict(self):
+        return dict(alphas=self.alphas)
+
+    def value(self, w):
+        """Compute the value of SLOPE at w."""
+        return np.sum(np.sort(np.abs(w)) * self.alphas[::-1])
+
+    def prox_1d(self, value, stepsize, j):
+        raise NotImplementedError(
```

Comment:
What do you think of ValueError ? Because  it's not that we did not implement it, it's that  it does not exist